test/e2e/cluster_upgrade_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// nolint: testpackage
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+var _ = Describe("Cluster upgrade test for k8s patch [K8s-upgrade]", func() {
+
+	var (
+		ctx                          context.Context
+		specName                     = "upgrade"
+		namespace                    *corev1.Namespace
+		cancelWatches                context.CancelFunc
+		clusterResources             *clusterctl.ApplyClusterTemplateAndWaitResult
+		dockerClient                 *client.Client
+		byoHostCapacityPool          = 4
+		byoHostName                  string
+		allbyohostContainerIDs       []string
+		allAgentLogFiles             []string
+		coreDNSUpgradeVersion        = "v1.8.6"
+		kubernetesVersionUpgradeTo   = "v1.23.5"
+		kubernetesVersionUpgradeFrom = "v1.23.4"
```

Comment:
Let's try a minor version upgrade? We already should have bundles available for `v1.22.x`
So let's change the test to upgrade from `v1.22.x` to `v1.23.y`

Reply:
Tested with minor version upgrade (v1.22.3 => v1.23.5) , it is working fine.